bin/www

```diff
@@ -19,10 +22,27 @@ var port = normalizePort(process.env.PORT || '3000');
 app.set('port', port);
 
 /**
- * Create HTTP server.
+ * Create HTTP or HTTPS server.
  */
 
-var server = http.createServer(app);
+var key_file_location = nconf.get('HTTPS_KEY_FILE');
+var cert_file_location = nconf.get('HTTPS_CERT_FILE');
+var server;
+//if both key and cert locations are set and they are valid file paths use them and run in https mode
+if (key_file_location !== undefined && cert_file_location !== undefined &&
+    fs.existsSync(key_file_location) && fs.existsSync(cert_file_location)) {
+    var options = {
+       key: fs.readFileSync(key_file_location),
+       cert: fs.readFileSync(cert_file_location)
+    };
+    server = https.createServer(options, app);
+    debug('Using, HTTPS_KEY_FILE = ' + key_file_location);
+    debug('Using, HTTPS_CERT_FILE = ' + cert_file_location);
+    debug('Server set to HTTPS mode');
+} else {
```

Comment:
If the vars are  set, but the paths are invalid, I think an exception should be thrown. Otherwise, a user might *think* he/she has things setup to be secure, but wind up running an insecure HTTP server instead